mainwindow.cpp

```diff
@@ -797,21 +799,26 @@ bool MainWindow::loadConfFile(const QString filename)
         while( !confFile.atEnd() )
         {
             enabledOption = true;
-            currentLine = confFile.readLine();
-            currentLine.remove(' ');
+            fullCurrentLine = confFile.readLine();
+            fullCurrentLine.remove(' ');
 
-            if( currentLine.startsWith("#@#") )
+            if( fullCurrentLine.startsWith("#@#") )
             {
-                currentLine.remove(0, 3);       //A line starting with extra_options_prefix is valid
+                fullCurrentLine.remove(0, 3);       //A line starting with extra_options_prefix is valid
                 enabledOption = false;
             }
 
-            if( !currentLine.startsWith('#') && currentLine.contains('=') )  //Ignore comments and lines without '='
+            if( !fullCurrentLine.startsWith('#') && fullCurrentLine.contains('=') )  //Ignore comments and lines without '='
             {
+                commentPosition = fullCurrentLine.indexOf('#');
+                if (commentPosition!=-1)
+                  currentLine = fullCurrentLine.left( commentPosition+1 );

```

Comment:
Why `+1` here?  Below there is a search for the character `=` and the following line does not use `+1`

Reply:
I see.  That's tricky.  Maybe it would be best to do the chop on the `fullCurrentLine` earlier and then you don't have to do this?

Even if you don't do this, the comment in chop line below is now wrong.

Reply:
fail to see why this is tricky - I tested and it works. Chopping first and then doing the rest will work, sure. Up to you to decide which way you go, but I see no problem with proposed silution, really. Yeah, the comment should say smth like «chop the trailing \n or #» in this case.

Reply:
I mean that it's tricky because it's not obvious in the reading why I would want to keep the `#` around.  Only after spending more time looking does the answer appear.  Code that is not obvious is harder to read.

I'm not very particular about it so if you don't want to change it, that's okay.